firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/impl/ApkUpdaterTest.java

```diff
@@ -202,7 +202,7 @@ public void updateApk_whenSuccessfullyUpdated_notificationsSetCorrectly()
 
     UpdateTask updateTask = apkUpdater.updateApk(TEST_RELEASE, true);
     List<UpdateProgress> events =
-        awaitProgressEvents(updateTask, 3, mockConnectionLatch::countDown);
+        awaitProgressEvents(updateTask, 3, countDownLatch::countDown);
```

Comment:
I think this is difficult to understand. It might make sense to split up the `awaitProgressEvents()` method to make this more readable. E.g.:

- declare the `progressEvents` lists here (inline)
- add the listener here (inline)
- count down the latch here (inline)
- call the truncated helper function: `awaitProgressEvents(progressEvents, 3)`

Reply:
Yes I like that a lot better, thanks!